lib/views/dock-item.js

```diff
@@ -109,6 +109,11 @@ export default class DockItem extends React.Component {
       const pane = this.props.workspace.paneForItem(this.dockItem);
       if (pane) {
         pane.activateItem(this.dockItem);
+        const dock = this.props.workspace.getPaneContainers()
+          .find(container => container.getPanes().find(p => p.getItems().includes(this.dockItem)));
+        if (dock && dock.constructor.name === 'Dock') {
```

Comment:
It might be better to probe for the method directly rather than using the constructor name:

```javascript
dock && dock.show && dock.show();
```